pkg/hooks/loader.go

```diff
@@ -174,14 +190,21 @@ func (h *Hook) SetTcsMocks(m []*models.Mock) {
 func (h *Hook) SetConfigMocks(m []*models.Mock) {
 	h.configMocks.deleteAll()
 	for index, mock := range m {
+		h.persistentMatchedMocks[mock.Name] = false
 		mock.TestModeInfo.SortOrder = index
 		mock.TestModeInfo.Id = index
 		h.configMocks.insert(mock.TestModeInfo, mock)
 	}
```

Comment:
why setting all to true ?

Reply:
I have added a comment describing the purpose of the key & value of `persistentMatchedMocks` 

https://github.com/keploy/keploy/blob/3faff242965634d853967ce4aed8182f9e33ad79/pkg/hooks/loader.go#L55


previously in test.go of service layer, i was fetching the matched mocks from array and resetting it. now, for persistentMatchedMocks to serve the purpose of removed `matchedMocks` array, I have implemented `getMatchedMocks` function. this function will look through the entries of persistentMatchedMocks and return the mocks which have value set to true (meaning it is not a config mock and it is not used till yet)
https://github.com/keploy/keploy/blob/3faff242965634d853967ce4aed8182f9e33ad79/pkg/service/test/util.go#L468